physics/SFC_Models/Land/Noahmp/noahmpdrv.F90

```diff
@@ -9,99 +9,469 @@
 !! as diagnotics calculation.
 
 !> This module contains the CCPP-compliant NoahMP land surface model driver.
-      module noahmpdrv
+  module noahmpdrv
 
-      use module_sf_noahmplsm
+    use module_sf_noahmplsm
+    
+    ! Land IAU increments for soil temperature (can also do soil moisture increments if needed)
+    use land_iau_mod,  only: land_iau_control_type, land_iau_external_data_type,   &
+                            land_iau_mod_set_control, land_iau_mod_init,          &
+                            land_iau_mod_getiauforcing, land_iau_mod_finalize,    &
+                            calculate_landinc_mask
 
-      implicit none
+    implicit none
 
-      integer, parameter :: psi_opt = 0 ! 0: MYNN or 1:GFS
+    integer, parameter :: psi_opt = 0 ! 0: MYNN or 1:GFS
 
-      private
+    private
 
-      public :: noahmpdrv_init, noahmpdrv_run
+    public :: noahmpdrv_init, noahmpdrv_run, &
+              noahmpdrv_timestep_init, noahmpdrv_timestep_finalize, noahmpdrv_finalize
```

Comment:
The timestep_finalize subroutinee can be removed since it is empty. One doesn't need to have subroutines present for CCPP phases that are not used in a scheme.